### PR TITLE
Cql json object api filter1

### PR DESCRIPTION
**Pull request by turingtestfail**
CQL-Object filter parsing integration.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

For all pull requests:

- [x ] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [ x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [ x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [x ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java [hunk 1]
```diff
@@ -17,35 +19,53 @@
 
     private static final FilterFactory2 FF = CommonFactoryFinder.getFilterFactory2();
     public static String CQL_TEXT = "cql-text";
+    public static String CQL_OBJECT = "cql-object";
 
     /**
-     * Parses the filter over the supported filter languages (right now, only {@link #CQL_TEXT}) and
-     * defaults the geometry liters in spatial filters to CRS84
+     * Parses the filter over the supported filter languages (right now, only {@link #CQL_TEXT} and
+     * {@link #CQL_OBJECT}) and defaults the geometry liters in spatial filters to CRS84
```
Comment (bradh) [inline src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java:26]: Not part of the change, but liters doesn't make sense in this sentence. Maybe geometry reference?

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java [hunk 2]
```diff
@@ -17,35 +19,53 @@
...
+        // right now there is a spec only for cql-text and cql-object, will be extended when more
+        // languages are
         // recognized (could have its own extension point too, if we want to allow easy extension
         // with new custom languages)
-        if (filterLang != null && !filterLang.equals(CQL_TEXT)) {
+        if (filterLang != null
+                && (!filterLang.equals(CQL_TEXT) && !filterLang.equals(CQL_OBJECT))) {
             throw new InvalidParameterValueException(
-                    "Only supported filter-lang at the moment is "
+                    "Only supported filter-lang options at the moment is "
```
Comment (bradh) [inline src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java:40]: is -> are?

**Review by bradh — COMMENTED** [review]
Looks like a nice bit of cleanup along the way too.

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java [hunk 3]
```diff
@@ -17,35 +19,53 @@
 
     private static final FilterFactory2 FF = CommonFactoryFinder.getFilterFactory2();
     public static String CQL_TEXT = "cql-text";
+    public static String CQL_OBJECT = "cql-object";
```
Comment (aaime) [inline src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java:22]: According to spec, should be "cql-json":

https://github.com/opengeospatial/ogcapi-features/blob/master/extensions/cql/standard/clause_6_filter.adoc#parameter-filter-lang

Looks like

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java (hunk 1 above)
Comment (aaime) [reply src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java:26]: It's a typo, was supposed to be ``literals``.

File: src/community/ogcapi/dggs/dggs-clickhouse/src/test/java/org/geotools/dggs/clickhouse/ClickHouseGeometrylessOnlineTest.java [hunk 4]
```diff
@@ -28,7 +28,7 @@ protected JDBCGeometrylessTestSetup createTestSetup() {
     }
 
     protected HashMap createDataStoreFactoryParams() throws Exception {
-        HashMap params = super.createDataStoreFactoryParams();
+        HashMap params = new HashMap(super.createDataStoreFactoryParams());
```
Comment (aaime) [inline src/community/ogcapi/dggs/dggs-clickhouse/src/test/java/org/geotools/dggs/clickhouse/ClickHouseGeometrylessOnlineTest.java:31]: This part conflicts with master, cannot merge because of it.

**Review by aaime — CHANGES_REQUESTED** [review]

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java (hunk 1 above)
Comment (turingtestfail) [reply src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java]: fixed

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java (hunk 2 above)
Comment (turingtestfail) [reply src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java]: fixed

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java (hunk 3 above)
Comment (turingtestfail) [reply src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/api/APIFilterParser.java]: fixed

File: src/community/ogcapi/dggs/dggs-clickhouse/src/test/java/org/geotools/dggs/clickhouse/ClickHouseGeometrylessOnlineTest.java (hunk 4 above)
Comment (turingtestfail) [reply src/community/ogcapi/dggs/dggs-clickhouse/src/test/java/org/geotools/dggs/clickhouse/ClickHouseGeometrylessOnlineTest.java]: fixed

**Comment by aaime** [comment]
Community modules get only the formatting bit of the QA checks, it's failing:

```
[ERROR] Failed to execute goal com.coveo:fmt-maven-plugin:2.4.0:check (default) on project gs-ogcapi-core: Found 1 non-complying files, failing build -> [Help 1]
```

Fixing is simple, just do a maven build before committing, it will format the files according to spec for you. Or, if you just want to run the formatting part, use ``mvn fmt:format``

**Comment by turingtestfail** [comment]
Formatted as requested.

**Review by aaime — APPROVED** [review]

**Comment by aaime** [comment]
Looks good, merging, thanks!
